src/output/plugins/PipeOutputPlugin.cxx

```diff
@@ -84,12 +97,30 @@ PipeOutput::Create(const ConfigBlock &block, Error &error)
 	return po;
 }
 
+#define set_MPDPIPE_vars(audio_format) {\
+	char strbuf[8];\
+	setenv("MPDPIPE_BITS", sample_format_to_string(audio_format.format), 1);\
+	snprintf(strbuf, sizeof(strbuf), "%u", audio_format.sample_rate);\
+	setenv("MPDPIPE_RATE", strbuf, 1);\
+	snprintf(strbuf, sizeof(strbuf), "%u", audio_format.channels);\
+	setenv("MPDPIPE_CHANNELS", strbuf, 1);\
+}
+
 inline bool
 PipeOutput::Open(AudioFormat &audio_format, Error &error)
 {
+#ifdef WIN32
+	set_MPDPIPE_vars(audio_format);
```

Comment:
This sets environment variables in the parent process, which I already told you is not acceptable. (Don't argue with me again. I don't accept it.)